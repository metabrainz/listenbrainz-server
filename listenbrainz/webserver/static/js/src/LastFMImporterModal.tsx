import * as React from "react";
import { faTimes } from "@fortawesome/free-solid-svg-icons";
import { IconProp } from "@fortawesome/fontawesome-svg-core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export type ModalProps = {
  disable: boolean;
  children: React.ReactElement[];
  onClose(event: React.MouseEvent<HTMLButtonElement>): void;
};

const LastFMImporterModal = (props: ModalProps) => {
  const divStyle = {
    position: "fixed",
    height: "auto",
    top: "50%",
    zIndex: 2,
    width: "90%",
    maxWidth: "500px",
    left: "50%",
    transform: "translate(-50%, -50%)",
    backgroundColor: "#fff",
    boxShadow: "0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22)",
    textAlign: "center",
    padding: "50px",
  } as React.CSSProperties;

  const buttonStyle = {
    position: "absolute",
    top: "5px",
    right: "10px",
    outline: "none",
    border: "none",
    background: "transparent",
  } as React.CSSProperties;

  const { children, onClose, disable } = props;

  return (
    <div style={divStyle} id="listen-progress-container">
      <button
        onClick={onClose}
        style={buttonStyle}
        disabled={disable}
        type="button"
      >
        <FontAwesomeIcon icon={faTimes as IconProp} />
      </button>
      <div>{children}</div>
    </div>
  );
};

export default LastFMImporterModal;
