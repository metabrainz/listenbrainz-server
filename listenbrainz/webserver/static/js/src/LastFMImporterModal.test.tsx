import * as React from "react";
import { mount, shallow } from "enzyme";

import LastFMImporterModal from "./LastFMImporterModal";

const props = {
  disable: false,
  children: [],
  onClose: (event: React.MouseEvent<HTMLButtonElement>) => {},
};

describe("LastFmImporterModal", () => {
  it("renders", () => {
    const wrapper = mount(<LastFMImporterModal {...props} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("close button is disabled/enabled based upon props", () => {
    const wrapper = shallow(<LastFMImporterModal {...props} />);
    // Test if close button is disabled
    wrapper.setProps({ disable: true });
    expect(wrapper.find("button").props().disabled).toBe(true);

    // Test if close button is enabled
    wrapper.setProps({ disable: false });
    expect(wrapper.find("button").props().disabled).toBe(false);
  });
});
