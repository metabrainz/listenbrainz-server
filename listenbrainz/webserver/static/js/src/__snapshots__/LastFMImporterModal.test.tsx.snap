// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastFmImporterModal renders 1`] = `"<div style=\\"position: fixed; height: auto; top: 50%; z-index: 2; width: 90%; max-width: 500px; left: 50%; transform: translate(-50%, -50%); background-color: rgb(255, 255, 255); box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22); text-align: center; padding: 50px;\\" id=\\"listen-progress-container\\"><button style=\\"position: absolute; top: 5px; right: 10px; outline: none; background: transparent;\\" type=\\"button\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"times\\" class=\\"svg-inline--fa fa-times fa-w-11 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 352 512\\"><path fill=\\"currentColor\\" d=\\"M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z\\"></path></svg></button><div></div></div>"`;
